File: 350. Intersection of Two Arrays II.py
class Solution:
    def intersect(self, nums1: List[int], nums2: List[int]) -> List[int]:
        nums1.sort()
        nums2.sort()
        
        i=0
        j=0
        arr = []
        
        while i<len(nums1) and j<len(nums2):
            if nums1[i]<nums2[j]:
                i=i+1
            elif nums1[i]>nums2[j]:
                j=j+1;
            elif nums1[i]==nums2[j]:
                arr.append(nums1[i])
                i=i+1
                j=j+1
        return arr
        

File: 350. Intersection of Two Arrays II.cpp
class Solution {
public:
    vector<int> intersect(vector<int>& nums1, vector<int>& nums2) {
        sort(nums1.begin(),nums1.end());
        sort(nums2.begin(),nums2.end());
        vector<int>v(min(nums1.size(),nums2.size()));
        auto it=set_intersection(nums1.begin(),nums1.end(),nums2.begin(),nums2.end(),v.begin());
        v.resize(it-v.begin());
        return v;
        
    }
};
